### PR TITLE
:bug: clusterawsadm: ensure role policies work

### PR DESCRIPTION
**Pull request by randomvariable**
**What this PR does / why we need it**:
Attempting to add extra statements for a real use case led to the CloudFormation template failing because it was missing the policy name. I had also accidentally appended these statements to the managed policy, which is not the documented intent.

This also adds an "AllowedEC2InstanceProfiles" field to be set for the controller to override which EC2 IAM roles can be used by Cluster API.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

Advice on testing strategies. Should I create a test directory with "correct" CloudFormation YAML's to compare the generator to?

### CODE REVIEW

**Comment by randomvariable** [comment]
tests added
/unhold

**Comment by randomvariable** [comment]
urgh typos
/hold

File: cmd/clusterawsadm/cloudformation/bootstrap/template_test.go [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+		tData, err := c.template().RenderCloudFormation().YAML()
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		if string(tData) != string(data) {
+			dmp := diffmatchpatch.New()
+			diffs := dmp.DiffMain(string(tData), string(data), false)
+			// out := string(tData)
```
Comment (detiber) [inline cmd/clusterawsadm/cloudformation/bootstrap/template_test.go:110]: Looks like left over from earlier iteration
```suggestion
```

**Pull request merged**